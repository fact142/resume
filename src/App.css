.resume{
  width: 30%;
  margin: 10px;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 1px solid #04f438;
  border-radius: 2px;
}
.resume input{
  margin: 10px;
}
.label{
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
