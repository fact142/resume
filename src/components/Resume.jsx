import React, { useState } from 'react';
import { deleteResume, putResume } from '../API/resumeAPI';
import { useDispatch } from 'react-redux';
import { updateResumes } from '../store/actions/resume';


const Resume = ({
                  id, firstname, lastname, age, gender, email, profile, address, phone, favoriteFilm, salary, about
                }) => {
  const [isEdit, setIsEdit] = useState(false);
  const [_id, setId] = useState(id);
  const [_firstname, setFirstname] = useState(firstname);
  const [_lastname, setLastname] = useState(lastname);
  const [_age, setAge] = useState(age);
  const [_gender, setGender] = useState(gender);
  const [_email, setEmail] = useState(email);
  const [_profile, setProfile] = useState(profile);
  const [_country, setCountry] = useState(address.country);
  const [_city, setCity] = useState(address.city);
  const [_phone, setPhone] = useState(phone);
  const [_favoriteFilm, setFavoriteFilm] = useState(favoriteFilm);
  const [_salary, setSalary] = useState(salary);
  const [_about, setAbout] = useState(about);




  const dispatch = useDispatch()
  const editResume = async () => {
    await putResume(
      _id, _firstname, _lastname, _age, _gender, _profile, _email,
      _country, _city, _phone, _favoriteFilm, _salary, _about,
    )
    dispatch(updateResumes())
    setIsEdit(!isEdit)
  }
  const removeResume = async () => {
    await deleteResume(id)
    dispatch(updateResumes())
  }

  return(
    (isEdit ?
        <div className="resume">
          <input value={_firstname} onChange={setFirstname} placeholder="Firstname" />
          <input value={_lastname} onChange={setLastname} placeholder="Lastname" />
          <input value={_age} onChange={setAge} placeholder="Age" />
          <input value={_gender} onChange={setGender} placeholder="Gender" />
          <input value={_email} onChange={setEmail} placeholder="Email" />
          <input value={_profile} onChange={setProfile} placeholder="Profile" />
          <input value={_country} onChange={setCountry} placeholder="Country" />
          <input value={_city} onChange={setCity} placeholder="City" />
          <input value={_phone} onChange={setPhone} placeholder="Phone" />
          <input value={_favoriteFilm} onChange={setFavoriteFilm} placeholder="Favorite Film" />
          <input value={_salary} onChange={setSalary} placeholder="Salary" />
          <input value={_about} onChange={setAbout} placeholder="Abount" />
        <button onClick={() => editResume()}>Сохранить</button>
      </div> :
        <div className="resume">
          <p>{'Имя: ' +  firstname}</p>
          <p>{'Фамилия: ' +  lastname }</p>
          <p>{'Возраст: ' +  age}</p>
          <p>{'Пол: ' +  gender}</p>
          <p>{'Почта ' +  email}</p>
          <p>{'Направление: ' +  profile}</p>
          <p>{'Адрес: ' +  address.country + ", " + address.city}</p>
          <p>{'Телефон: ' +  phone}</p>
          <p>{'Любимый фильм: ' +  favoriteFilm}</p>
          <p>{'З/п: ' +  salary}</p>
          <p>{'О себе: ' +  about}</p>
          <button onClick={() => setIsEdit(!isEdit)}>Редактировать</button>
          <button onClick={() => removeResume()}>Удалить</button>
        </div>)

  )
}

export default Resume;
