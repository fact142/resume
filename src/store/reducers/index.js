import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import resumeReducer from './resumeReducer'

const rootReducer = combineReducers({
  resume: resumeReducer,
});
const composedEnhancer = composeWithDevTools(applyMiddleware(thunk));
const store = createStore(rootReducer, composedEnhancer);

export default store;
