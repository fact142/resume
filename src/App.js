
import './App.css';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import Main from './components/Main';
import { Error } from './components/Error';
function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path="/main" component={Main} />
        <Redirect exact strict from="/" to="main" />
        <Route path="/" component={Error} />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
