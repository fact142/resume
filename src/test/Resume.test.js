import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Resume from '../components/Resume';
import { Provider } from 'react-redux';
import store from '../store/reducers';
import App from '../App';

const test = {
  id:1,
  firstname: 'David',
  lastname: 'Lynch',
  age: 75,
  gender: 'MALE',
  email: 'twinpeaks@lora.com',
  profile: 'Genius',
  address: {
    country: 'Nobody',
    city: 'Knows'
  },
  phone: '89825',
  favoriteFilm: 'Iron Man 2',
  salary: 99,
  about: 'Love weather reports'
}

const TestingFormButton = () =>(<Provider store={store}> <Resume
  id={test.id}
  firstname={test.firstname}
  lastname={test.lastname}
  age={test.age}
  gernder={test.gender}
  email={test.email}
  profile={test.profile}
  address={test.address}
  phone={test.phone}
  favoriteFilm={test.favoriteFilm}
  salary={test.salary}
  about={test.about}
  />
</Provider>)


describe('Resume component', () => {
  //Тестирование снимками
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <TestingFormButton />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  //Проверка на рендер кнопки
  it('should render button from props', () => {
    const { getByText } = render(
      <TestingFormButton />,
    )
    expect(getByText("Редактировать")).toBeInTheDocument();
  });
});
