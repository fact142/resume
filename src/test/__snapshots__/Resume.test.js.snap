// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume component should render correctly 1`] = `
Array [
  " ",
  <div
    className="resume"
  >
    <p>
      Имя: David
    </p>
    <p>
      Фамилия: Lynch
    </p>
    <p>
      Возраст: 75
    </p>
    <p>
      Пол: undefined
    </p>
    <p>
      Почта twinpeaks@lora.com
    </p>
    <p>
      Направление: Genius
    </p>
    <p>
      Адрес: Nobody, Knows
    </p>
    <p>
      Телефон: 89825
    </p>
    <p>
      Любимый фильм: Iron Man 2
    </p>
    <p>
      З/п: 99
    </p>
    <p>
      О себе: Love weather reports
    </p>
    <button
      onClick={[Function]}
    >
      Редактировать
    </button>
    <button
      onClick={[Function]}
    >
      Удалить
    </button>
  </div>,
]
`;
